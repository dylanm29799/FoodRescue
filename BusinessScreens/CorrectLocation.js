/*
 *
 * ClassName: CorrectLocation.js
 *
 * Date: 28/04/2021
 *
 *
 * @author: Dylan Murphy, X17506166
 *
 * @reference : https://www.udemy.com/course/react-native-the-practical-guide/learn/lecture/15674818?start=0#overview
 * @reference : https://docs.expo.io/
 * @reference : https://firebase.google.com/docs/web/setup
 * @reference : https://github.com/wix/react-native-navigation
 * @reference : https://stackoverflow.com/questions/15017052/understanding-email-validation-using-javascript
 *
 */
import React, { useState } from "react";
import { Text, View, StyleSheet, TouchableOpacity } from "react-native";
import { TextInput } from "react-native-gesture-handler";
import { scale } from "../components/ResponsiveText";
import Colour from "../constants/Colour";
import Geocoder from "react-native-geocoding";
import MapView, { Marker } from "react-native-maps";
import * as firebase from "firebase";
import "firebase/firestore";

const CorrectLocation = (props) => {
  //Firebase link
  const dbconnection = firebase.firestore();
  //Geocoder API Key
  Geocoder.init("AIzaSyAZHSAnfYc7nm3UbyJtJr_NgLIKmZf-Tfk", { language: "en" });
  //User details
  var user = firebase.auth().currentUser;
  //State for geocode location
  const [geoCode, setGeoCode] = useState("");
  //Example location for map
  const [lng, setLng] = useState(-6.2603);
  const [lat, setLat] = useState(53.3498);
  //Geocoder - Take location from state and set the longitude and latitude of the area that matches the best with the geocode provided
  const runGeocode = () => {
    Geocoder.from(geoCode)
      .then((json) => {
        var location = json.results[0].geometry.location;
        console.log(location);
        setLat(location.lat);
        setLng(location.lng);
      })
      .catch((error) => console.warn(error));
  };

  const runUpdate = () => {
    //Update the location in firebase
    dbconnection
      .collection("businessDetails")
      .doc(user.uid)
      .get()
      .then(function (doc) {
        if (doc.exists) {
          console.log("Success");
          return dbconnection
            .collection("businessDetails")
            .doc(user.uid)
            .update({ longitude: lng, latitude: lat });
        }
      })
      .then(function () {
        console.log("Navigating");
        //Navigate to the business home
        props.navigation.navigate({ routeName: "BusinessHome" });
      });
  };

  return (
    <View style={styles.Screen}>
      <Text style={styles.text}>Please enter your address here:</Text>
      <TextInput
        style={styles.textInput}
        placeholder="18 City Quay, Dublin"
        onChangeText={setGeoCode}
      ></TextInput>

      <TouchableOpacity onPress={runGeocode} style={styles.touchable}>
        <Text style={styles.touchableText}>Check</Text>
      </TouchableOpacity>

      <MapView
        style={{ width: "80%", height: "30%" }}
        region={{
          longitude: lng,
          latitude: lat,
          latitudeDelta: 0.02,
          longitudeDelta: 0.0,
        }}
      >
        <Marker
          coordinate={{ latitude: lat, longitude: lng }}
          pinColor={Colour.primaryColour}
          title={"Business Name"}
        />
      </MapView>
      <Text style={styles.text}>
        Once you're happy with your address, click Submit
      </Text>
      <Text style={{ paddingVertical: scale(2) }}></Text>
      <TouchableOpacity onPress={runUpdate} style={styles.touchable}>
        <Text style={styles.touchableText}>Submit</Text>
      </TouchableOpacity>
    </View>
  );
};
//Stylesheet for styling
const styles = StyleSheet.create({
  Screen: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  text: {
    fontSize: scale(16),
    fontFamily: "MonM",
    textAlign: "center",
    maxWidth: "80%",
    paddingTop: scale(30),
  },
  textInput: {
    borderBottomColor: Colour.primaryColour,
    borderBottomWidth: 2,
    width: "80%",
    marginVertical: scale(10),
    paddingTop: scale(10),
    textAlign: "center",
    fontFamily: "MonM",
  },
  touchable: {
    height: scale(30),
    width: scale(70),
    backgroundColor: Colour.primaryColour,
    justifyContent: "center",
    marginBottom: scale(15),
    marginHorizontal: scale(10),
  },
  touchableText: {
    color: "#fff",
    fontFamily: "MonB",

    textAlign: "center",
    fontSize: scale(12),
  },
});

export default CorrectLocation;
