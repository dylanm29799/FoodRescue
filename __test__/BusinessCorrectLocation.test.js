/*
 *
 * ClassName: // __tests__/BusinessCorrectLocation.test.js

 *
 * Date: 28/04/2021
 *
 *
 * @author: Dylan Murphy, X17506166
 *
 * @reference : https://docs.expo.io/guides/testing-with-jest/
 * @reference : https://medium.com/react-native-training/learning-to-test-react-native-with-jest-part-1-f782c4e30101
 *
 */

import React from "react";
import CorrectLocation from "../BusinessScreens/CorrectLocation";
import renderer from "react-test-renderer";
import { firebaseConfig } from "../config";
import * as firebase from "firebase";

firebase.initializeApp(firebaseConfig);

jest.mock("react-navigation", () => ({
  withNavigation: (Component) => (props) => (
    <Component navigation={{ navigate: jest.fn() }} {...props} />
  ),
}));

test("renders correctly", () => {
  const tree = renderer.create(<CorrectLocation />).toJSON();
  expect(tree).toMatchSnapshot();
});
