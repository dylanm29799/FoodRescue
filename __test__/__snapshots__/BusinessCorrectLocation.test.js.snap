// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontFamily": "MonM",
        "fontSize": 34.285714285714285,
        "maxWidth": "80%",
        "paddingTop": 64.28571428571428,
        "textAlign": "center",
      }
    }
  >
    Please enter your address here:
  </Text>
  <TextInput
    allowFontScaling={true}
    collapsable={false}
    onChangeText={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    placeholder="18 City Quay, Dublin"
    rejectResponderTermination={true}
    style={
      Object {
        "borderBottomColor": "#c44536",
        "borderBottomWidth": 2,
        "fontFamily": "MonM",
        "marginVertical": 21.428571428571427,
        "paddingTop": 21.428571428571427,
        "textAlign": "center",
        "width": "80%",
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#c44536",
        "height": 64.28571428571428,
        "justifyContent": "center",
        "marginBottom": 32.14285714285714,
        "marginHorizontal": 21.428571428571427,
        "opacity": 1,
        "width": 150,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "MonB",
          "fontSize": 25.714285714285715,
          "textAlign": "center",
        }
      }
    >
      Check
    </Text>
  </View>
  <AIRMap
    handlePanDrag={false}
    initialRegion={null}
    onChange={[Function]}
    onLayout={[Function]}
    onMapReady={[Function]}
    onMarkerPress={[Function]}
    region={
      Object {
        "latitude": 53.3498,
        "latitudeDelta": 0.02,
        "longitude": -6.2603,
        "longitudeDelta": 0,
      }
    }
    style={
      Object {
        "height": "30%",
        "width": "80%",
      }
    }
  >
    <AIRMapMarker
      coordinate={
        Object {
          "latitude": 53.3498,
          "longitude": -6.2603,
        }
      }
      onPress={[Function]}
      pinColor="#c44536"
      stopPropagation={false}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "position": "absolute",
          },
          undefined,
        ]
      }
      title="Business Name"
    />
  </AIRMap>
  <Text
    style={
      Object {
        "fontFamily": "MonM",
        "fontSize": 34.285714285714285,
        "maxWidth": "80%",
        "paddingTop": 64.28571428571428,
        "textAlign": "center",
      }
    }
  >
    Once you're happy with your address, click Submit
  </Text>
  <Text
    style={
      Object {
        "paddingVertical": 4.285714285714286,
      }
    }
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#c44536",
        "height": 64.28571428571428,
        "justifyContent": "center",
        "marginBottom": 32.14285714285714,
        "marginHorizontal": 21.428571428571427,
        "opacity": 1,
        "width": 150,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "MonB",
          "fontSize": 25.714285714285715,
          "textAlign": "center",
        }
      }
    >
      Submit
    </Text>
  </View>
</View>
`;
